Clean up the README

TeknikalDomain opened the pull request:
Some general cleanup of README.

Parts have been re-written to improve readability, or remove clutter and awkward English.

TeknikalDomain committed:
Clean up the README

2ndBillingCycle commented on README.md:
````diff
@@ -56,23 +59,23 @@ To match something with spaces, it must be wrapped in parenthesis:
 ```
 
-This goes for the `[server name]` and `#[channel name]` as well, as those both are treated as patterns, too.
+This applies the same for `[server name]` and `#[channel name]` as well.
````
```
This also applies to the `[server name]` and `#[channel name]`.
```

2ndBillingCycle commented on README.md:
```diff
@@ -56,23 +59,23 @@ To match something with spaces, it must be wrapped in parenthesis:
-To have Lua treat them as regular characters, put a `%` before them.
+To have Lua treat any of these as regular characters, put a `%` before them.
 
 As an example, to match the text `:-) tehee (-:`, the whole thing first needs to be wrapped in parenthesis, and the `(`, `)`, and `-` inside need to be escaped with a `%`:
```
```
As an example, to match the text `:-) tehee (-:`, the `[pattern]` first needs to be wrapped in parentheses, and the `(`, `)`, and `-` that are part of the `[pattern]` need to be escaped with a `%`:
```

TeknikalDomain replied:
References to `[pattern]` here aren't useful.

2ndBillingCycle commented on README.md:
````diff
@@ -98,7 +101,7 @@ If the command is typed into a channel, either the `[server name]`, the `[channe
 ```
 
-This is convenient, because the plugin only sees the [hostname][] (like a URL) of the specific server that's hosting the IRC channel, and not what might show up in the HexChat Network List.
+This is convenient, because the plugin only sees the [hostname][] (Server URL) of the specific server that's hosting the IRC channel, and not what might show up in the HexChat Network List.
````
How would I indicate that the hostname HexChat uses is the one reported by the server?

For example, `test.example.com` can be reported by the server, but won't be how the server is identified *anywhere* in HexChat's UI, as the server was connected to by IP.

TeknikalDomain replied:
Added an information section at the end of the README with an overview.

2ndBillingCycle reviewed (changes requested):
Thank you for cleaning up the [readme](./README.md). I think the proposed changes help make the document more readable.

Could the comments I left on some of the changes be addressed?

TeknikalDomain committed:
Slightly improve readability, explain hostnames, address PR concerns

TeknikalDomain committed:
Shameless self-promotion / in-joke removed, explain hostnames even more

2ndBillingCycle committed:
Updated README.me

 - Inlined images
 - Changed command syntax to match HexChat's conventions
 - Made sure # is prepended to all [channel name]
 - Made example IRC sessions easier to follow.
 - Added section that details acceptable audio file formats

TeknikalDomain committed:
General cleanup of README, fix typos

TeknikalDomain committed:
Cleanup README headers, convert to title-case

TeknikalDomain commented on README.md:
```diff
@@ -214,17 +216,14 @@ This means that the following will not trigger a sound:
-* Signed 16-bit PCM
-* Signed 16-bit PCM
-* Signed 16-bit PCM
 * Signed 16-bit PCM
```
All these were identical. Unless there was some reason for listing the same item four times, I removed the duplicates.

2ndBillingCycle replied:
Thank you. I meant to list 16-, 24-, 32-bit.

TeknikalDomain commented on README.md:
````diff
@@ -288,13 +287,13 @@ If a `Channel message` is received that reads `ATTENTION: All staff report to Sa
 
 If HexChat is started more than once, and this plugin is loaded in each instance, settings set in any HexChat window are saved, but those changes aren't picked up by the other clients.
 
-Reload the plugin to update thaat window:
+Reload the plugin to update that window:
 
 ```
 /lua reload specialsounds.lua
 ```
 
-Using a different file name if it's not installed to HexChat's `config/addons/` folder.
+Use a different file name if it's not installed to HexChat's `config/addons/` folder.
````
Elaborate?

2ndBillingCycle replied:
I meant to address the scenario where HexChat cannot locate the plugin.

TeknikalDomain replied:
Yes, but "Use a different filename" doesn't seem to be that helpful... as in it gives almost 0 direction what to actually do, other than "do something else"

2ndBillingCycle replied:
How does the following sound, inserted in the [Installation](https://github.com/2ndBillingCycle/specialsounds/blob/7196708d51ca2c52979c4dd800aedebac2087ac3/README.md#installation) section?

---

Sometimes, HexChat does not find the file:

```
Lua syntax error: cannot open C:\hexchat\config\addons\specialsound.lua: No such file or directory
```

If this happens, or if `SpecialSounds.lua` is not downloaded to the `config/addons/` directory, give the full path to the file. If this has spaces, wrap the path in `"`:

```
/lua load C:\Users\2nd\Downloads\SpecialSounds.lua
```

TeknikalDomain replied:
Perfect.

TeknikalDomain commented on README.md:
```diff
@@ -195,15 +195,15 @@ This can be tedious, but hopefully doesn't come up much. If it does, and you wan
 # Limitations
 
-## Available message types
+## Available Message Types
```
Heading title might not fit with rest of section

2ndBillingCycle replied:
## Only Channel Messages Are Matched

## Limited Message Types

Do either of these sound better?

TeknikalDomain replied:
"Only Channel Messages Are Matched" is personally what I would go with

TeknikalDomain commented on README.md:
```diff
@@ -214,17 +216,14 @@ This means that the following will not trigger a sound:
-* Signed 16-bit PCM
 * Signed 16-bit PCM
 * 32-bit float
 * μ-Law
```
u-law and a-law are both raw format audio. Consider removing, even if accurate?

2ndBillingCycle replied:
While in large part the sound data is the same,

![Comparison of file contents between WAV and raw](https://user-images.githubusercontent.com/51795309/67800375-27e51700-fa55-11e9-9010-815630682a70.png)

[`PlaySound`](https://docs.microsoft.com/en-us/previous-versions/dd743680(v%3Dvs.85)) does not play the file without the WAV header.

2ndBillingCycle replied:
Additionally, both algorithms compress the data as compared to standard Signed 16-bit PCM:

![Size comparison of the same data in FLAC, μ-Law, and Signed 16-bit PCM](https://user-images.githubusercontent.com/51795309/67801108-852d9800-fa56-11e9-9c79-4b3a3b3879cd.png)

TeknikalDomain replied:
You're not wrong. but they also lack any form of appreciable header, it mostly just dumps the raw audio data onto disk... and the same in reverse. I can open literally any file on my computer as a valid u-law / a-law and it'll import correctly.

2ndBillingCycle replied:
Unfortunately, [`PlaySound`](https://docs.microsoft.com/en-us/previous-versions/dd743680(v%3Dvs.85)) requires files to be structured like [`WAVEFORMATEX`](https://docs.microsoft.com/en-us/previous-versions/dd757713(v=vs.85)) or similar.

It doesn't take up much space to list them, and they're in the screenshot.

2ndBillingCycle committed:
Corrected typo and link

2ndBillingCycle committed:
Updated list of acceptable WAV filetypes
